proposals/lip-0057.md

```diff
@@ -1056,19 +1057,32 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
 
 To finalize the state of the genesis block the following logic is executed.  If any step fails, the block is discarded and has no further effect.
 
-As in the previous point, let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the DPoS module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisDPoSStoreSchema` schema, given above. Then:
+As in the previous point, let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the DPoS module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisDPoSStoreSchema` schema, given above. Moreover, let `b` be the genesis block. Then:
 
 ```python
-# Register all validators in the Validators module
-for validator in genesisBlockAssetObject.validators:
-    validators.registerValidatorKeys(
-        validator.address,
-        validator.proofOfPossession,
-        validator.generatorKey,
-        validator.blsKey
-    )
-    if the above returns False:
-        fail
+# Register all validators in the Validators module.
+# For the snapshot block for the migration from Lisk Core 3 to Lisk Core 4 on 
+# Lisk mainnet, apply a special rule. This must be done as validators in 
+# this snapshot block do not have a BLS key.
+if the chain is the mainchain and b is the snapshot block for the migration:
```

Comment:
This will trigger if chain is mainchain but b is not the snapshot block. Maybe you wanted to assign snapshot block to a variable b somewhere else? Because I don't also see `b` is used in below code.

Reply:
I don't think it's clear, since I ask a question. I understood it now though, can be resolved.